types/src/sample.rs

```diff
@@ -210,6 +210,20 @@ impl From<Sample> for RawSample {
     }
 }
 
+impl TryFrom<RawSample> for Sample {
+    type Error = Error;
+
+    fn try_from(raw: RawSample) -> std::result::Result<Self, Self::Error> {
+        let share = raw.share.ok_or(Error::MissingShares)?.try_into()?;
```

Comment:
this won't work for parity shares, and there is again not much information to judge if share should be treated as parity or not